File: app/src/main/java/com/live/utils/NetworkUtil.java
package com.live.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.live.LiveApp;

public class NetworkUtil {

    /**
     * 检查网络是否可用
     */
    public static boolean isNetworkAvailable() {
        ConnectivityManager manager = (ConnectivityManager) LiveApp.getInstance().getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        if (manager == null) {
            return false;
        }
        NetworkInfo networkinfo = manager.getActiveNetworkInfo();
        return networkinfo != null && networkinfo.isAvailable();
    }

    /**
     * 判断WIFI网络是否可用
     */
    public static boolean isWifiConnected() {
        ConnectivityManager connectivityManager = (ConnectivityManager) LiveApp.getInstance().getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivityManager == null) {
            return false;
        }
        NetworkInfo activeNetInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetInfo != null&& activeNetInfo.getType() == ConnectivityManager.TYPE_WIFI;

    }


}

File: app/src/main/java/com/live/utils/NetworkUtil.kt
package com.live.utils

import android.content.Context
import android.net.ConnectivityManager
import com.live.LiveApp

object NetworkUtil {

    /**
     * 检查网络是否可用
     */
    val isNetworkAvailable: Boolean
        get() {
            val manager = LiveApp.instance!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val networkinfo = manager.activeNetworkInfo
            return networkinfo != null && networkinfo.isAvailable
        }

    /**
     * 判断WIFI网络是否可用
     */
    val isWifiConnected: Boolean
        get() {
            val connectivityManager = LiveApp.instance!!.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetInfo = connectivityManager.activeNetworkInfo
            return activeNetInfo != null && activeNetInfo.type == ConnectivityManager.TYPE_WIFI

        }


}
